site/content/blog/2020-06-04-svelte-and-typescript.md

```diff
@@ -0,0 +1,123 @@
+
+The Svelte compiler support for TypeScript is handled by [Christian Kaisermann](https://github.com/kaisermann)'s [`svelte-preprocess`](https://github.com/sveltejs/svelte-preprocess#svelte-preprocess) which is now an official Svelte project.
+
+For the editor level, we took inspiration from [Pine's](https://github.com/octref) work in the [Vue](https://vuejs.org) ecosystem via [Vetur](https://github.com/vuejs/vetur). Vetur provides an [LSP](https://github.com/vuejs/vetur/blob/master/server), a VS Code extension and a [CLI](https://github.com/vuejs/vetur/blob/master/vti). Svelte now also has an [LSP](https://github.com/sveltejs/language-tools/blob/master/packages/language-server), a [VS Code extension](https://github.com/sveltejs/language-tools/blob/master/packages/svelte-vscode) and a [CLI](https://github.com/sveltejs/language-tools/blob/master/packages/svelte-check).
+
+
+#### `*.svelte` Introspection
+
+For the official Svelte VS Code extension, we built off the foundations which [James Birtles](https://github.com/UnwrittenFun) has created in [`UnwrittenFun/svelte-vscode`](https://github.com/UnwrittenFun/svelte-vscode) and [`UnwrittenFun/svelte-language-server`](https://github.com/UnwrittenFun/svelte-language-server/).
+
+[Simon Holthausen](https://github.com/dummdidumm) and [Lyu, Wei-Da](https://github.com/jasonlyu123) have done great work improving the JavaScript and TypeScript introspection, including integrating [@halfnelson](https://github.com/halfnelson)'s [svelte2tsx](https://github.com/sveltejs/language-tools/tree/master/packages/svelte2tsx#svelte2tsx) which powers understanding the props on components in your codebase.
+
+
+### Try today
+
+The Svelte template has been extended with a script to convert it to a TypeScript project.
+
+```bash
+npx degit sveltejs/template svelte-app
+cd svelte-app
+node scripts/updateTypeScriptVersion.js
+```
+
+### Adding TypeScript support to your project
+
+Before getting started, add the dependencies:
+
+```bash
+yarn add --dev svelte-preprocess @rollup/plugin-typescript @tsconfig/svelte svelte-check typescript
+npm install --save-dev svelte-preprocess @rollup/plugin-typescript @tsconfig/svelte svelte-check typescript
+```
+
+##### 1. Compiling TypeScript
+
+You first need to set up [`svelte-preprocess`](https://github.com/sveltejs/svelte-preprocess#svelte-preprocess) and  [`@rollup/plugin-typescript`](https://github.com/rollup/plugins/tree/master/packages/typescript#rollupplugin-typescript) for the compiler:
+
+When in a rollup project, this would mean:
+
+```diff
++ import autoPreprocess from 'svelte-preprocess'
```

Comment:
```suggestion
+ import sveltePreprocess from 'svelte-preprocess';
```

Reply:
This is to keep it inline with the svelte-preprocess docs: https://github.com/sveltejs/svelte-preprocess#what-is-it

![Screen Shot 2020-07-17 at 5 05 51 PM](https://user-images.githubusercontent.com/49038/87830770-c61d4500-c84f-11ea-87a2-113ca70cb1ff.png)

Reply:
I'm open to changing that 😁  I've already named the default export `preprocess`, `autoPreprocess`, and `sveltePreprocess`. My only concern is blurring the line between the `preprocess` property and `svelte-preprocess` even more. I also thought about exporting an `auto` preprocessor instead of using the default export (keeping the default for compatibility, of course).

Reply:
I don't have a strong opinion other than consistency 👍

Reply:
`sveltePreprocess` makes more sense to me

Reply:
Will change in `svelte-preprocess` then!